api/src/main/java/edu/wpi/first/shuffleboard/api/plugin/Plugin.java

```diff
@@ -26,21 +28,73 @@
+ *
+ * <p>For the same reasons, a plugin that depends on another <i>must</i> provide that information with a
+ * {@link Requires @Requires} or {@link Requirements @Requirements} annotation. The former is a <i>repeatable</i>
+ * annotation for which the latter is a wrapper; as such, it is recommended to use {@code @Requires} annotations to
+ * increase readability.
  */
 public class Plugin {
 
   private final String groupId;
   private final String name;
-  private final String version;
-  private final String description;
+  private final Version version;
+  private final String summary;
   private final BooleanProperty loaded = new SimpleBooleanProperty(this, "loaded", false);
 
-  protected Plugin(String groupId, String name, String version, String description) {
-    this.groupId = groupId;
-    this.name = name;
-    this.version = version;
-    this.description = description;
+  /**
+   * Creates a new plugin instance. The subclass <i>must</i> have a {@link Description @Description} annotation that
+   * defines the group ID, name, version, and summary of the plugin. If the plugin depends on another (eg the camera
+   * server plugin depends on the network tables one), that should be specified with a {@link Requires @Requires}
+   * annotation.
+   *
+   * @throws InvalidPluginDefinitionException if no {@code @Description} annotation is present, or if the
+   */
+  protected Plugin() {
+    validatePluginClass(getClass());
+    Description description = getClass().getAnnotation(Description.class);
+
+    this.groupId = description.group();
+    this.name = description.name();
+    this.version = Version.parse(description.version());
+    this.summary = description.summary();
+  }
+
+  /**
+   * Validates that a plugin class has valid description and dependency annotations.
+   *
+   * @param pluginClass the plugin class to validate
+   *
+   * @throws InvalidPluginDefinitionException if the class has an invalid definition
+   */
+  public static void validatePluginClass(Class<? extends Plugin> pluginClass) {
+    Description description = pluginClass.getAnnotation(Description.class);
+    if (description == null) {
+      throw new InvalidPluginDefinitionException(
+          "The plugin class does not have a @Description annotation: " + pluginClass.getName());
+    }
+    String group = description.group();
+    if (group.contains(":")) {
+      throw new InvalidPluginDefinitionException("The group ID cannot contain a colon (:) character: " + group);
+    }
+    String name = description.name();
+    if (name.contains(":")) {
+      throw new InvalidPluginDefinitionException("The plugin name cannot contain a colon (:) character: " + name);
+    }
+    String version = description.version();
+    try {
+      Version.parse(version);
+    } catch (IllegalArgumentException e) {
+      throw new InvalidPluginDefinitionException("The version string does not follow semantic versioning", e);
+    }
```

Comment:
You might want to create one uber exception with all of the problems rolled up into one exception instead of getting each one in this order. This will help developers to resolve each problem without having to re-launch to see if anything else is wrong.